core/lib/instrument.c

```diff
@@ -371,6 +371,19 @@ DECLARE_CXTSWPROT_VAR(static mutex_t client_aux_lib64_lock,
                       INIT_LOCK_FREE(client_aux_lib64_lock));
 #endif
 
+#if defined(STATIC_LIBRARY) && !defined(WINDOWS)
+// To support static DR used for both standalone and clients we need to provide
+// some copy of client main. There is no WEAK on Windows so we do not support
+// this usage there.
+WEAK void
+dr_client_main(client_id_t id, int argc, const char *argv[])
+{
+    // This will be called when using static DR but no client, so we can't
```

Comment:
Is it reasonable to assume that now the following will never be NULL?

https://github.com/DynamoRIO/dynamorio/blob/839a2c380c9ce82311d0cad46337c7020cfb94c3/core/lib/instrument.c#L735C16-L735C20

Reply:
No, it will still fail as before for regular use with a client that has no dr_client_main routine: it's querying the client library, which is normally separate from the DR library where this new weak symbol was added.  Even for DR+client linked into the app, it will fail for pure-static apps b/c the lookup uses dlsym which fails for pure-static.